gemstone/src/swapper/jupiter/provider.rs

```diff
@@ -94,14 +110,32 @@ impl GemSwapProvider for Jupiter {
         let route = &quote.data.routes[0];
         let quote_response: QuoteResponse = serde_json::from_str(&route.route_data).map_err(|_| SwapperError::InvalidRoute)?;
         let fee_mint = self.get_fee_mint(&quote.request.mode, &quote_response.input_mint, &quote_response.output_mint);
-        let fee_account = self.get_fee_account(&quote.request.options, &fee_mint);
+        let mut fee_account = self.get_fee_account(&quote.request.options, &fee_mint);
```

Comment:
can you make sure this called only if neither input/output can't be used for referral fee, then use 0 referral fee and allow them to swap it, to avoid any extra requests

Reply:
Done